pkg/cacheutil/memcached_server_selector.go

```diff
@@ -30,9 +31,8 @@ var (
 // with consistent DNS names where the naturally sorted order
 // is predictable (ie. Kubernetes statefulsets).
 type MemcachedJumpHashSelector struct {
-	// To avoid copy and pasting all memcache server list logic,
-	// we embed it and implement our features on top of it.
-	servers memcache.ServerList
```

Comment:
Would it be simple to add a new method to the custom selector for getting bucketed keys? 

```
func (s *MemcachedJumpHashSelector) GetServersForKeys(keys) (bucketed map[string][]string, err error) {
  addrs := *(addrsPool.Get().(*[]net.Addr))
  err := s.servers.Each(func(addr net.Addr) error {
    addrs = append(addrs, addr)
    return nil
  })

  // Rest of the logic 
  
}
```

Reply:
But the interface we have is this and we cannot add/change existing interface otherwise `ServerList` doesn't implement the interface

```
type updatableServerSelector interface {
	memcache.ServerSelector

	// SetServers changes a ServerSelector's set of servers at runtime
	// and is safe for concurrent use by multiple goroutines.
	//
	// SetServers returns an error if any of the server names fail to
	// resolve. No attempt is made to connect to the server. If any
	// error occurs, no changes are made to the internal server list.
	SetServers(servers ...string) error
}
```